auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1492,6 +1492,9 @@ private void transition() {
         }
 
         private InitialState selectMechanismConfiguration() {
+            if (mechanismConfigurationSelector == null) {
+                throw log.mechanismConfigurationSelectorNotDefined();
```

Comment:
If the mechanismConfigurationSelector cannot be null we can checkNotNullParam in the constructor instead.